spec/unit/network/http/factory_spec.rb

```diff
@@ -54,16 +53,16 @@ def create_connection(site)
     it 'should not set a proxy if a no_proxy env var matches the destination' do
       Puppet[:http_proxy_host] = proxy_host
       Puppet[:http_proxy_port] = proxy_port
-      expect(Puppet::Util::HttpProxy).to receive(:no_proxy?).and_return(true)
-      conn = create_connection(site)
+      Puppet::Util.withenv('NO_PROXY' => site.host) do
```

Comment:
Can you also make a test where the withenv is used to set a successful no_proxy?

Reply:
Oops - got the sense of the test backwards!